core/hotspot/cache/count_min_sketch.go

```diff
@@ -0,0 +1,85 @@
+package cache
+
+const sketchDepth = 4
+
+// countMinSketch is an implementation of count-min sketch with 4-bit counters.
+type countMinSketch struct {
+	counters []uint64
+	mask     uint32
+}
+
+// init initialize count-min sketch with the given width.
+func newCountMinSketch(width int) *countMinSketch {
+	size := nextPowerOfTwo(uint32(width)) >> 2
+	if size < 1 {
+		size = 1
+	}
+	countMinSketch := countMinSketch{
+		make([]uint64, size),
+		size - 1,
+	}
+	return &countMinSketch
+}
+
+// add increase counters with given hash
+func (c *countMinSketch) add(h uint64) {
+	hash1, hash2 := uint32(h), uint32(h>>32)
+
+	for i := uint32(0); i < sketchDepth; i++ {
+		combinedHash := hash1 + (i * hash2)
+		idx, off := c.position(combinedHash)
+		c.inc(idx, (16*i)+off)
+	}
+}
+
+// estimate returns minimum value of counters associated with the given hash.
+func (c *countMinSketch) estimate(h uint64) uint8 {
+	hash1, hash2 := uint32(h), uint32(h>>32)
+
+	var min uint8 = 0xFF
+	for i := uint32(0); i < sketchDepth; i++ {
+		combinedHash := hash1 + (i * hash2)
+		idx, off := c.position(combinedHash)
+		count := c.val(idx, (16*i)+off)
+		if count < min {
+			min = count
+		}
+	}
+	return min
+}
+
+func (c *countMinSketch) reset() {
+	for i, v := range c.counters {
+		if v != 0 {
+			//divides all by two.
+			c.counters[i] = (v >> 1) & 0x7777777777777777
```

Comment:
Use const instead of magic number "0x7777777777777777"

Reply:
done